### PR TITLE
Signup: Show domain step in import flows

### PR DESCRIPTION
**Pull request by creativecoder**
#### Changes proposed in this Pull Request

* Show domain step in `import` and `import-onboarding` flows (no longer autocomplete it)
* Autofill domain input with a suggested domain import site url
    - If there's a path in the url, such as `https://hi6822.wixsite.com/eat-here-its-good`, suggest the path `eat-here-its-good`
    - Else if its a WordPress.com, Blogger, GoDaddy, Squarespace, or Tumblr site, such as `http://theperfectrealestate.godaddysites.com/`, suggest the subdomain `theperfectrealestate`
    - Otherwise suggest the domain, like `example.com` or `blog.example.com`

Screen Shot 2019-08-07 at 17 12 18

##### Import flow

- Visit `/start/import`
- Complete the user step, if logged out
- Enter a Wix or GoDaddy site url
- When you get to the domains step, you should see a suggested domain created from the site url you entered in the previous step and a search for an available domain should start automatically
- Select a free or paid domain and continue, confirming that you can complete the flow and create a new site

##### Domains flow

Domains only flows also search automatically in the domains step, so verify that's still working

- Visit `/start/domain/domain-only?new=testtesttesttest&search=yes`
- Verify that `testtesttesttest` is filled in the search input a search for an available domain starts automatically

### CODE REVIEW

**Comment by creativecoder** [comment]
Pinging @aidvu , who worked on domains-only auto searching, to make sure I haven't missed anything else related to this functionality.

File: client/signup/steps/domains/index.jsx [hunk 1]
```diff
@@ -85,27 +85,13 @@ class DomainsStep extends React.Component {
...
-				stepName: props.stepName,
-				stepSectionName: props.stepSectionName,
-			} );
-			props.goToNextStep();
-			return;
-		}
-
 		const domain = get( props, 'queryObject.new', false );
 		const search = get( props, 'queryObject.search', false ) === 'yes';
+		const suggestedDomain = get( props, 'signupDependencies.suggestedDomain' );
```
Comment (aidvu) [inline client/signup/steps/domains/index.jsx]: Doing things in the constructor like this is very brittle. Please test all the cases when landing on the domain screen with the `suggestedDomain` set like refreshing the browser, going back and forth through the steps.

File: client/lib/importers/utils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	const pathname = parsedUrl.pathname ? trim( parsedUrl.pathname, '/' ) : null;
+
+	// Site name is sometimes the path, as in the case of Wix and Medium sites,
+	// such as `medium.com/my-blog`
+	if ( pathname ) {
+		return pathname.replace( '/', '-' );
+	}
+
+	// Other services use a subdomain, like `example.wordpress.com.
+	if ( parsedHostname && parsedHostname.length > 2 ) {
```
[the remark this answers is not in the document]
Comment (andfinally) [reply client/lib/importers/utils.ts]: Just discovered `url.parse` ignores the `www`, so this works fine!

**Review by andfinally — COMMENTED** [review]
This LGTM, I've tested with a variety of URLs in both logged-out and logged-in states, and using the browser back button and flow back button, and everything works as expected. We're getting eight tests failing - I'm not sure why, for example:

```
"after all" hook - [WPCOM] Calypso Gutenberg Editor: Pages (desktop) Insert a payment button into a page: @parallel
invalid session id

"after all" hook - [WPCOM] Editor: Pages (desktop) Insert a payment button into a page: @parallel @jetpack
invalid session id

"after all" hook - [WPCOM] Editor: Posts (desktop) Insert a payment button: @parallel @jetpack
invalid session id
```

**Review by andfinally — APPROVED** [review]
Tests have now passed, so apart from resolving the conflict I'd say this is good to go!

File: client/lib/signup/step-actions.js [hunk 3]
```diff
@@ -192,6 +185,11 @@ export function createSiteWithCart(
 		newSiteParams.find_available_url = !! isPurchasingItem;
 	}
 
+	if ( 'import' === lastKnownFlow || 'import-onboarding' === lastKnownFlow ) {
```
[the remark this answers is not in the document]
Comment (creativecoder) [reply client/lib/signup/step-actions.js]: I'm not totally sure either, but in testing adding the plans step to the `import` flow, we saw `flowName` empty in some cases, so it seems more reliable to use `lastKnownFlow`

**Comment by creativecoder** [comment]
I've rebased this and updated the domain suggestion logic. Now subdomain is used as a suggestion only for specific services, WordPress.com, Blogger, GoDaddy, Squarespace, and Tumblr. Otherwise we suggest the full domain, so the suggestion for `blog.example.com` isn't `blog`, it's `blog.example.com`.

**Review by jonathansadowski — CHANGES_REQUESTED** [review]
Can we move the `client/lib/importers` code into `client/lib/importer`? I'm attempting to make that change in #35240, as I feel the distinction between the two isn't strong enough to warrant the two confusingly similar names

**Review by jonathansadowski — CHANGES_REQUESTED** [review]
Edit: False alarm, testing error.

**Comment by creativecoder** [comment]
> Can we move the client/lib/importers code into client/lib/importer? I'm attempting to make that change in #35240, as I feel the distinction between the two isn't strong enough to warrant the two confusingly similar names

Good call -- updated.

**Review by jonathansadowski — APPROVED** [review]
Works well for me. Thanks for making the change to the library path! Goodjob

**Pull request merged**